From 2fa22b867b4ac5c2771789d959e1a5943620d789 Mon Sep 17 00:00:00 2001
From: Emil Medve <Emilian.Medve@Freescale.com>
Date: Sat, 18 Jan 2014 05:34:07 -0600
Subject: [PATCH 072/466] fsl_qbman: Use 'unsigned' for always positive
 numbers

This fixes the following sparse warnings:

warning: incorrect type in argument 2 (different signedness)
   expected int *cpu
   got unsigned int *<noident>
warning: incorrect type in argument 2 (different signedness)
   expected unsigned int [usertype] *flags
   got int *<noident>

Signed-off-by: Emil Medve <Emilian.Medve@Freescale.com>
Change-Id: I06de94a56bf65f299a68275161445868f1fa526e
Reviewed-on: http://git.am.freescale.net:8181/8101
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Haiying Wang <Haiying.Wang@freescale.com>
Reviewed-by: Emilian Medve <Emilian.Medve@freescale.com>
[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/staging/fsl_qbman/dpa_sys.h        |    2 +-
 drivers/staging/fsl_qbman/qman_test_high.c |    3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_qbman/dpa_sys.h b/drivers/staging/fsl_qbman/dpa_sys.h
index 59e88c8..43651f3 100644
--- a/drivers/staging/fsl_qbman/dpa_sys.h
+++ b/drivers/staging/fsl_qbman/dpa_sys.h
@@ -246,7 +246,7 @@ static inline type *name##_find(struct dpa_rbtree *tree, u32 val) \
  * range of cpu indices, and each item optionally be prefixed by "s" to indicate
  * that the portal associated with that cpu should be shared. See bman_driver.c
  * for more specifics. */
-static int __parse_portals_cpu(const char **s, int *cpu)
+static int __parse_portals_cpu(const char **s, unsigned int *cpu)
 {
 	*cpu = 0;
 	if (!isdigit(**s))
diff --git a/drivers/staging/fsl_qbman/qman_test_high.c b/drivers/staging/fsl_qbman/qman_test_high.c
index 833beb8..caba5b3 100644
--- a/drivers/staging/fsl_qbman/qman_test_high.c
+++ b/drivers/staging/fsl_qbman/qman_test_high.c
@@ -130,7 +130,8 @@ static void do_enqueues(struct qman_fq *fq)
 
 void qman_test_high(void)
 {
-	int flags, res;
+	unsigned int flags;
+	int res;
 	struct qman_fq *fq = &fq_base;
 
 	pr_info("qman_test_high starting\n");
-- 
1.7.10.4

